client-tez/src/main/java/org/apache/tez/runtime/library/common/shuffle/impl/RssShuffleManager.java

```diff
@@ -0,0 +1,1236 @@
+              .setNameFormat("ShuffleRunner {" + srcNameTrimmed + "}")
+              .build());
+      Future reporterFuture = reporterExecutor.submit(new ReporterCallable());
+    }
+
+    ListenableFuture<Void> runShuffleFuture = schedulerExecutor.submit(rssSchedulerCallable);
+    Futures.addCallback(runShuffleFuture, new SchedulerFutureCallback(), MoreExecutors.directExecutor());
+    // Shutdown this executor once this task, and the callback complete.
+    schedulerExecutor.shutdown();
+  }
+
+  private class ReporterCallable extends CallableWithNdc<Void> {
+    /**
+     * Measures if the batching interval has ended.
+     */
+
+    ReporterCallable() {
+    }
+
+    @Override
+    protected Void callInternal() throws Exception {
+      long nextReport = 0;
+      while (!isShutdown.get()) {
+        try {
+          reportLock.lock();
+          while (failedEvents.isEmpty()) {
+            boolean signaled = reportCondition.await(maxTimeToWaitForReportMillis,
+                TimeUnit.MILLISECONDS);
+          }
+
+          long currentTime = Time.monotonicNow();;
+          if (currentTime > nextReport) {
+            if (failedEvents.size() > 0) {
+              List<Event> failedEventsToSend = Lists.newArrayListWithCapacity(
+                  failedEvents.size());
+              for (InputReadErrorEvent key : failedEvents.keySet()) {
+                failedEventsToSend.add(InputReadErrorEvent
+                    .create(key.getDiagnostics(), key.getIndex(),
+                        key.getVersion()));
+              }
+              inputContext.sendEvents(failedEventsToSend);
+              failedEvents.clear();
+              nextReport = currentTime + maxTimeToWaitForReportMillis;
+            }
+          }
+        } finally {
+          reportLock.unlock();
+        }
+      }
+      return null;
+    }
+  }
+
+
+  private boolean isAllInputFetched() {
+    LOG.info("Check isAllInputFetched, numNoDataInput:{}, numWithDataInput:{},numInputs:{},  " 
+            + "successRssPartitionSet:{},  allRssPartition:{}.",
+        numNoDataInput, numWithDataInput, numInputs, successRssPartitionSet, allRssPartition);
+    return (numNoDataInput.get() + numWithDataInput.get() >= numInputs)
+        && (successRssPartitionSet.size() >= allRssPartition.size());
+  }
+
+  private boolean isAllInputAdded() {
+    LOG.info("Check isAllInputAdded, numNoDataInput:{}, numWithDataInput:{},numInputs:{},  "
+            + "successRssPartitionSet:{}, allRssPartition:{}.",
+        numNoDataInput, numWithDataInput, numInputs, successRssPartitionSet, allRssPartition);
+    return numNoDataInput.get() + numWithDataInput.get() >= numInputs;
+  }
+
+  private class RssRunShuffleCallable extends CallableWithNdc<Void> {
+
+    private final Configuration conf;
+
+    RssRunShuffleCallable(Configuration conf) {
+      this.conf = conf;
+    }
+
+    @SuppressWarnings("RV_RETURN_VALUE_IGNORED_BAD_PRACTICE")
```

Comment:
Could we solve this issue instead of ignoring it?

Reply:
ok， I solved it and is now all compiled ok.